bigtable/tests/filters_integration_test.cc

```diff
@@ -0,0 +1,682 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <google/protobuf/text_format.h>
+#include <cmath>
+#include <sstream>
+#include "bigtable/admin/admin_client.h"
+#include "bigtable/admin/table_admin.h"
+#include "bigtable/client/cell.h"
+#include "bigtable/client/data.h"
+#include "bigtable/client/filters.h"
+
+namespace btproto = ::google::bigtable::v2;
+
+namespace {
+void ReportException();
+
+// TODO(#32) - change the function signature when Table::ReadRows() is a thing.
+// All these functions would become:
+//     `Function(bigtable::Table& table, std::string const& row_key)`
+//     `Function(bigtable::Table& table, std::string const& begin,
+//               std::string const& end)`
+//
+
+void CheckPassAll(bigtable::Client& client, bigtable::Table& table,
+                  std::string const& row_key);
+void CheckBlockAll(bigtable::Client& client, bigtable::Table& table,
+                   std::string const& row_key);
+void CheckLatest(bigtable::Client& client, bigtable::Table& table,
+                 std::string const& row_key);
+void CheckFamilyRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckColumnRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckColumnRange(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckTimestampRange(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key);
+void CheckCellsRowLimit(bigtable::Client& client, bigtable::Table& table,
+                        std::string const& row_key_prefix);
+void CheckCellsRowOffset(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key_prefix);
+void CheckCellsRowSample(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key_prefix);
+}  // anonymous namespace
+
+int main(int argc, char* argv[]) try {
+  namespace admin_proto = ::google::bigtable::admin::v2;
+
+  // Make sure the arguments are valid.
+  if (argc != 4) {
+    std::string const cmd = argv[0];
+    auto last_slash = std::string(argv[0]).find_last_of("/");
+    std::cerr << "Usage: " << cmd.substr(last_slash + 1)
+              << " <project> <instance> <table>" << std::endl;
+    return 1;
+  }
+  std::string const project_id = argv[1];
+  std::string const instance_id = argv[2];
+  std::string const table_name = argv[3];
+  std::string const family = "fam";
```

Comment:
What's the naming convention here? `family`, `fam0`, etc.? Why not just fam 0 -> 3?

It looks like `family` is somehow different than the rest, but doesn't appear so from the code below.

Reply:
Good point.  Fixed.